azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -107,32 +102,36 @@ def __init__(self, name, path_maker, limits=None):
                 self.disable()
                 raise CGroupsException("Hierarchy {0} is not mounted".format(hierarchy))
 
-            cgroup_name = "" if self.is_wrapper_cgroup else self.name
-            cgroup_path = path_maker(hierarchy, cgroup_name)
+            cgroup_path = path_maker(hierarchy, self.name)
             if not os.path.isdir(cgroup_path):
                 CGroupConfigurator._try_mkdir(cgroup_path)
                 logger.info("Created cgroup {0}".format(cgroup_path))
 
             self.cgroups[hierarchy] = cgroup_path
 
+    _is_systemd_return_value = None
+
     @staticmethod
-    def is_systemd_manager():
+    def is_systemd():
         """
-        Determine if systemd is managing system services. Many extensions are structured as a set of services,
-        including the agent itself; systemd expects those services to remain in the cgroups in which it placed them.
-        If this process (presumed to be the agent) is in a cgroup that looks like one created by systemd, we can
-        assume systemd is in use.
+        Determine if systemd is managing system services. If this process (presumed to be the agent) is in a CPU cgroup
+        that looks like one created by systemd, we can assume systemd is in use.
+
+        NOTE: This function has the side effect of mounting the cgroups file system if it is not already mounted.
```

Comment:
When does this happen? I don't see where the mounting happens here.

Reply:
comment was for a previous iteration of the code... i'll remove it